### ISSUE TITLE
Problem with keyterm annotation without -multiThread option

### ISSUE THREAD

**Issue by kermitt2** [anHALytics/anHALytics-core#76]
Behavior of annotateAllKeyTerm is different with -multiThread option - it works fine - and without - documents are incorrectly considered as already annotated, except fist one.  

Here example of 8 documents loaded without annotations for keyterm: 

```bash
lopez@work:~/anHALytics-core/anhalytics-annotate$ java -Xmx2048m -jar target/anhalytics-annotate-0.1-SNAPSHOT.one-jar.jar -exe annotateAllKeyTerm
log4j:WARN No such property [datePattern] in org.apache.log4j.ConsoleAppender.
[INFO ] fr.inria.anhalytics.commons.managers.MongoManager: Mongodb is running on server : localhost port : 27017
[INFO ] fr.inria.anhalytics.commons.managers.MongoManager: Database : anhalytics
[INFO ] fr.inria.anhalytics.annotate.services.AnnotateService: Checking KEYTERM service...
[INFO ] fr.inria.anhalytics.annotate.services.AnnotateService: KEYTERM  service is ok and can be used.
[INFO ] fr.inria.anhalytics.commons.managers.MongoFileManager: 8 documents found for : 2010-06-13
[INFO ] fr.inria.anhalytics.annotate.Annotator: processing teis for :2010-06-13
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 main Start. Processing = hal-00297631
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 main: hal-00297631 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 main End. :694 ms
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-00317008: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-00320274: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-00499067: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-00563000: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-00651627: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping hal-01070389: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: skipping inserm-00504792: already annotated
[INFO ] fr.inria.anhalytics.annotate.Annotator: Total: 1 documents annotated.
```

```
lopez@work:~/anHALytics-core/anhalytics-annotate$ java -Xmx2048m -jar target/anhalytics-annotate-0.1-SNAPSHOT.one-jar.jar -multiThread -exe annotateAllKeyTerm
log4j:WARN No such property [datePattern] in org.apache.log4j.ConsoleAppender.
[INFO ] fr.inria.anhalytics.commons.managers.MongoManager: Mongodb is running on server : localhost port : 27017
[INFO ] fr.inria.anhalytics.commons.managers.MongoManager: Database : anhalytics
[INFO ] fr.inria.anhalytics.annotate.services.AnnotateService: Checking KEYTERM service...
[INFO ] fr.inria.anhalytics.annotate.services.AnnotateService: KEYTERM  service is ok and can be used.

[INFO ] fr.inria.anhalytics.commons.managers.MongoFileManager: 8 documents found for : 2010-06-13
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-1 Start. Processing = hal-00297631
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-2 Start. Processing = hal-00317008
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-3 Start. Processing = hal-00320274
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-4 Start. Processing = hal-00499067
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-2: hal-00317008 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-2 End. :573 ms
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-2 Start. Processing = hal-00563000
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-3: hal-00320274 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-3 End. :648 ms
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-3 Start. Processing = hal-00651627
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-1: hal-00297631 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-1 End. :813 ms
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-1 Start. Processing = hal-01070389
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-4: hal-00499067 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-4 End. :900 ms
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-4 Start. Processing = inserm-00504792
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-2: hal-00563000 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-2 End. :461 ms
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-3: hal-00651627 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-3 End. :521 ms
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-4: inserm-00504792 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-4 End. :383 ms
[INFO ] fr.inria.anhalytics.annotate.KeyTermAnnotatorWorker: 		 pool-1-thread-1: hal-01070389 annotated by the KeyTerm extraction and disambiguation service.
[INFO ] fr.inria.anhalytics.annotate.AnnotatorWorker: 		 pool-1-thread-1 End. :503 ms
[INFO ] fr.inria.anhalytics.annotate.Annotator: Finished all threads
[INFO ] fr.inria.anhalytics.annotate.Annotator: Total: 8 documents annotated.
```

**Comment by kermitt2** [comment]
It's actually the case for all annotators apparently.

**Comment by kermitt2** [comment]
This is due to getHalIDFromHalDocID() in Utilities which return an empty string if the halID does not contain a version number. It should return the halID if no version number is appended.

**Comment by kermitt2** [comment]
fix with commit 78f680801aa75cef56807278ff2efc76d3281753
